File: spring_java_orm_sample/src/main/java/com/example/demo/jooq/dsl/StoreDsl.java
package com.example.demo.jooq.dsl;

import lombok.RequiredArgsConstructor;
import org.jooq.DSLContext;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;

import static com.example.demo.jooq.generated.Tables.STORE;

@Component
@Profile("jooq")
@RequiredArgsConstructor
public class StoreDsl {
    private final DSLContext dsl;

    public boolean exists(Long storeId) {
        return dsl.fetchExists(STORE, STORE.ID.eq(storeId));
    }
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/jooq/dsl/StoreDsl.kt
package com.example.demo.jooq.dsl

import com.example.demo.jooq.generated.Tables
import org.jooq.DSLContext
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component

@Component
@Profile("jooq")
class StoreDsl(private val dsl: DSLContext) {
    fun exists(storeId: Long?): Boolean {
        return dsl.fetchExists(Tables.STORE, Tables.STORE.ID.eq(storeId))
    }
}
